From: Bhavna Sarathy <bnagendr@redhat.com>
Date: Wed, 18 Aug 2010 15:08:10 -0400
Subject: [xen] allow dom0 to control core performance boost
Message-id: <20100818151401.8436.60128.sendpatchset@localhost.localdomain>
Patchwork-id: 27681
O-Subject: [RHEL5.6 XEN PATCH] Allow dom0 to control core performance boot
Bugzilla: 568771
RH-Acked-by: Paolo Bonzini <pbonzini@redhat.com>
RH-Acked-by: Andrew Jones <drjones@redhat.com>

Resolves BZ 568771

This patch allows dom0 to control core performance boost.  This patch opens
up the MSR that dom0 needs to write to in order to enable or disable core
performance boost.

While the upstream implementation has 3 patches as listed below, the RHEL5
Xen HV has never done cpufreq management, and has used the powernow driver
in dom0.  So other than allowing dom0 to control CPB, there isn't much to do.

Upstream CPB changesets:
http://xenbits.xensource.com/xen-unstable.hg?rev/6c3db6c83a02
http://xenbits.xensource.com/xen-unstable.hg?rev/b432d7d9be92
http://xenbits.xensource.com/xen-unstable.hg?rev/b44a4f9b9a62

Please review and ACK

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/x86/traps.c b/arch/x86/traps.c
index 4b6e6db..0aeeeff 100644
--- a/arch/x86/traps.c
+++ b/arch/x86/traps.c
@@ -1850,6 +1850,7 @@ static int emulate_privileged_op(struct cpu_user_regs *regs)
         case MSR_K8_PSTATE5:
         case MSR_K8_PSTATE6:
         case MSR_K8_PSTATE7:
+        case MSR_K8_HWCR:
             if ( boot_cpu_data.x86_vendor != X86_VENDOR_AMD )
                 goto fail;
             if ( !is_cpufreq_controller(v->domain) )
